client/my-sites/domains/domain-management/list/bulk-edit-contact-info.jsx

```diff
@@ -0,0 +1,218 @@
+import { errorNotice, infoNotice, successNotice } from 'calypso/state/notices/actions';
+import { UPDATE_CONTACT_INFORMATION_EMAIL_OR_NAME_CHANGES } from 'calypso/lib/url/support';
+import FormLabel from 'calypso/components/forms/form-label';
+import FormCheckbox from 'calypso/components/forms/form-checkbox';
+import DesignatedAgentNotice from 'calypso/my-sites/domains/domain-management/components/designated-agent-notice';
+import { isFetchingUserSettings } from 'calypso/state/user-settings/selectors';
+
+import wp from 'calypso/lib/wp';
+
+const wpcom = wp.undocumented();
+
+/**
+ * Style dependencies
+ */
+import './list-all.scss';
+
+class BulkEditContactInfo extends React.Component {
+	static propTypes = {
+		handleSaveContactInfo: PropTypes.func.isRequired,
+		isDisabled: PropTypes.bool,
+		isSubmitting: PropTypes.bool,
+		onTransferLockOptOutChange: PropTypes.func.isRequired,
+		emailOnly: PropTypes.bool,
+	};
+
+	static defaultProps = {
+		isDisabled: false,
+		isSubmitting: false,
+		emailOnly: false,
+	};
+
+	state = {
+		hasSetContactDetailsFromCache: false,
+		contactDetails: {},
+		errorMessages: {},
+	};
+
+	componentDidMount() {
+		if ( isEmpty( this.props.userSettings ) ) {
+			this.props.fetchUserSettings();
+		}
+
+		if ( isEmpty( this.props.contactDetailsCache ) ) {
+			this.props.requestContactDetailsCache();
+		}
+
+		this.debouncedValidateContactDetails = debounce( this.validateContactDetails, 500 );
+	}
+
+	componentDidUpdate() {
+		const email = ! isEmpty( this.props.userSettings?.new_user_email )
+			? this.props.userSettings?.new_user_email
+			: this.props.userSettings?.user_email;
+
+		if ( ! isEmpty( email ) && isEmpty( this.state.contactDetails?.email ) ) {
+			this.updateDomainContactFields( { email: email } );
+		}
+
+		if (
+			! isEmpty( this.props.contactDetailsCache ) &&
+			false === this.state.hasSetContactDetailsFromCache
+		) {
+			this.setContactDetailsFromCache( this.props.contactDetailsCache );
+		}
+	}
+
+	isLoading() {
+		return (
+			isEmpty( this.props.contactDetailsCache ) ||
+			this.props.isRequestingContactDetailsCache ||
+			this.props.isFetchingUserSettings
+		);
+	}
+
+	setContactDetailsFromCache = ( data ) => {
+		delete data.email;
+		this.setState( { hasSetContactDetailsFromCache: true }, () => {
+			this.updateDomainContactFields( data );
+		} );
+	};
+
+	updateDomainContactFields = ( data ) => {
+		const newContactDetails = merge( {}, this.state.contactDetails, data );
+		this.setState( { contactDetails: newContactDetails } );
+		! isEmpty( newContactDetails ) && this.debouncedValidateContactDetails( newContactDetails );
+	};
+
+	onTransferLockOptOutChange = ( event ) => {
+		this.props.onTransferLockOptOutChange( event.target.checked );
+	};
+
+	validateContactDetails = ( contactDetails ) => {
+		if ( this.isLoading() ) {
+			return;
+		}
+
+		wpcom.validateDomainContactInformation( contactDetails, [], ( error, data ) => {
```

Comment:
Will this have any effect if there are no domain names specified? Maybe it will... I just wanted to check.

Reply:
It does. But it won't take into account any of the TLD-specific rules. But since those rules don't apply to the email field, I am not including them here.

Reply:
Eh.  I'm actually going to go ahead and add the selected domains in here to prevent future-me from having forgetting about this.